graspologic/layouts/render.py

```diff
@@ -0,0 +1,314 @@
+    node_colors: Dict[Any, str],
+    vertex_alpha: float,
+    edge_line_width: float,
+    edge_alpha: float,
+    figure_width: float,
+    figure_height: float,
+    vertex_line_width: float = 0.01,
+    vertex_shape: str = "o",
+    arrows: bool = False,
+    dpi: int = 100,
+):
+    if len(positions) != len(graph.nodes()):
+        raise ValueError(
+            f"The number of positions provided {len(positions)} is not the same as the "
+            f"number of nodes in the graph {len(graph.nodes())}"
+        )
+    for position in positions:
+        if position.node_id not in graph:
+            raise ValueError(
+                f"The node position provided for {position.node_id} references a node "
+                f"not found in our graph"
+            )
+
+    plt.rcParams["figure.dpi"] = dpi  # TODO, test at different dpi
+
+    plt.clf()
+    figure = plt.gcf()
+    ax = plt.gca()
+    ax.set_axis_off()
+    figure.set_size_inches(figure_width, figure_height)
+    window_extent_width = ax.get_window_extent().width
+
+    x_domain, y_domain = _calculate_x_y_domain(positions)
+
+    position_map = {position.node_id: position for position in positions}
+    node_positions = {
+        position.node_id: (position.x, position.y) for position in positions
+    }
+
+    vertices = []
+    vertex_sizes = []
+    node_color_list = []
+    edge_color_list = []
+
+    for node in graph.nodes():
+        vertices.append(node)
+        vertex_sizes.append(position_map[node].size)
+        node_color_list.append(node_colors[node])
+
+    vertex_sizes = _scale_node_sizes_for_rendering(
+        vertex_sizes, x_domain, (0, window_extent_width), dpi
+    )
+
+    for source, target in graph.edges():
+        edge_color_list.append(node_colors[source])
+
+    ax.set_xbound(x_domain)
+    ax.set_xlim(x_domain)
+    ax.set_ybound(y_domain)
+    ax.set_ylim(y_domain)
+
+    nx.draw_networkx_edges(
+        graph,
+        pos=node_positions,
+        alpha=edge_alpha,
+        width=edge_line_width,
+        edge_color=edge_color_list,
+        arrows=arrows,
+        ax=ax,
+    )
+
+    nx.draw_networkx_nodes(
+        graph,
+        pos=node_positions,
+        nodelist=vertices,
+        node_color=node_color_list,
+        alpha=vertex_alpha,
+        linewidths=vertex_line_width,
+        node_size=vertex_sizes,
+        node_shape=vertex_shape,
+        ax=ax,
+    )
+
+
+def show_graph(
+    graph: nx.Graph,
+    positions: List[NodePosition],
+    node_colors: Dict[Any, str],
+    vertex_line_width: float = 0.01,
+    vertex_alpha: float = 0.55,
+    edge_line_width: float = 0.5,
+    edge_alpha: float = 0.02,
+    figure_width: float = 15.0,
+    figure_height: float = 15.0,
+    light_background: bool = True,
+    vertex_shape: str = "o",
+    arrows: bool = False,
+    dpi: int = 500,
+):
+    """
+    Renders and displays a graph.
+
+    Attempts to display it via the platform-specific display library such as TkInter
```

Comment:
So *draw* literally means it opens up a window and shows it to you, at that moment.  Matplotlib usually (always?) uses TkInter to do this, but you need to install it on some systems to make it work - I'm pretty sure on Ubuntu you literally have to install tkinter as an `apt` thing not just a straight python thing, since it's actually displaying it as a GUI type window for you and lets you zoom and pan and save and whatnot.  It's different on MacOS and Windows, but I don't know the super specifics.

The main point is that you shouldn't use this function unless you have a platform-specific display library like tkinter installed.